cloud/linode/instances.go

```diff
@@ -29,7 +64,33 @@ func (e instanceNoIPAddressesError) Error() string {
 	return fmt.Sprintf("instance %d has no IP addresses", e.id)
 }
 
+func (i *instances) linodeByName(nodeName types.NodeName) (*linodego.Instance, error) {
+	i.nodeCache.RLock()
+	defer i.nodeCache.RUnlock()
+	for _, node := range i.nodeCache.nodes {
+		if node.Label == string(nodeName) {
+			return node, nil
+		}
+	}
+
+	return nil, cloudprovider.InstanceNotFound
+}
+
+func (i *instances) linodeByID(id int) (*linodego.Instance, error) {
+	i.nodeCache.RLock()
+	defer i.nodeCache.RUnlock()
+	instance, ok := i.nodeCache.nodes[id]
+	if !ok {
+		return nil, cloudprovider.InstanceNotFound
+	}
+	return instance, nil
+}
+
 func (i *instances) lookupLinode(ctx context.Context, node *v1.Node) (*linodego.Instance, error) {
+	if err := i.nodeCache.refresh(ctx, i.client); err != nil {
```

Comment:
So if a new node gets added, it will take 90 seconds for the nodeCache to update with the new node? That seems a little long, is there a reason for that number?

Reply:
The duration is fairly arbitrary. I wanted all the subsequent calls to hit the cache, but they do happen in succession, so we could shorten this quite a bit. I thought we were constrained by the 5-minute refresh intervals anyway, so 90 seconds seemed like a good middle ground. I'll shorten it to 15 seconds and I'll grep logs to see how much our calls to the API are bundled together.

Reply:
The 5 minute interval is only for updating the metadata, but when nodes are created or shutdown, that 5 minute interval does not come into play. The node monitor period which is involved in checking if a node is up is being increased from 5 seconds to 10 seconds which I think will interact with the cache more. So 15 seconds would work, or we could do 30.